src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -99,8 +99,12 @@ class Definitions {
   lazy val RootPackage: TermSymbol = ctx.newSymbol(
     NoSymbol, nme.ROOTPKG, PackageCreationFlags, TypeRef(NoPrefix, RootClass))
 
-  lazy val EmptyPackageClass = ctx.newCompletePackageSymbol(RootClass, nme.EMPTY_PACKAGE).moduleClass.asClass
-  lazy val EmptyPackageVal = EmptyPackageClass.sourceModule.entered
+  lazy val EmptyPackageVal = ctx.newCompletePackageSymbol(RootClass, nme.EMPTY_PACKAGE).entered
+  lazy val EmptyPackageClass = EmptyPackageVal.moduleClass.asClass
+
+  /** A package in which we can place all methods that are interpreted specially by the compiler */
```

Comment:
@sjrd can you call to javascript from ScalaJS? Then you can just remove special handling of it entirely. Indeed, `throw` method implementation cannot be written in scala, but you already intercept some methods, eg identityHashCode as far as I know.

@odersky, in case of `throw` why is it required to treat it specially by compiler? Can't it be normal call to normal java method, which will just `throw` the Throwable that was given to it?
This will allow to remove of special handling of `throw` in JVM backend. But array ops indeed need special handling.